Controls/Input/Input.UWP/AutoCompleteBox/AutoCompleteTextBox.cs

```diff
@@ -17,6 +25,36 @@ public AutoCompleteTextBox()
             this.DefaultStyleKey = typeof(AutoCompleteTextBox);
         }
 
+        internal void UpdateDeleteButtonVisibility()
+        {
+            if (string.IsNullOrEmpty(this.Text))
+            {
+                this.deleteButton.Visibility = Visibility.Collapsed;
+                return;
+            }
+
+            this.deleteButton.Visibility = this.isClearButtonVisible == true ? Visibility.Visible : Visibility.Collapsed;
+        }
+
+        private void AutoCompleteTextBox_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            this.UpdateDeleteButtonVisibility();
+        }
+
```

Comment:
Private methods should be below the protected ones or the StyleCop will throw error.